src/Promitor.Scraper.Host/Validation/Steps/MetricsDeclarationValidationStep.cs

```diff
@@ -99,11 +98,11 @@ private static List<string> ValidateMetrics(List<MetricDefinition> metrics, Metr
             errorMessages.AddRange(metricErrorMessages);
 
             // Detect duplicate metric names
-            var duplicateMetricNames = DetectDuplicateMetrics(metrics);
-            foreach (var duplicateMetricName in duplicateMetricNames)
-            {
-                errorMessages.Add($"Metric name '{duplicateMetricName}' is declared multiple times");
-            }
+
+            errorMessages.AddRange(
+                DetectDuplicateMetrics(metrics)
+                    .Select(duplicateMetricName =>
+                        $"Metric name '{duplicateMetricName}' is declared multiple times"));
```

Comment:
I prefer to not have inline calls so would love to use this:

```suggestion
var duplicateMetrics = DetectDuplicateMetrics(metrics);
errorMessages.AddRange(duplicateMetrics.Select(duplicateMetricName => $"Metric name '{duplicateMetricName}' is declared multiple times"));
```